src/components/modals/WalletListModal2.js

```diff
@@ -0,0 +1,250 @@
+// @flow
+
+import { FormField, MaterialInputStyle } from 'edge-components'
+import React, { Component, Fragment } from 'react'
+import { FlatList, View } from 'react-native'
+
+import { CryptoExchangeCreateWalletRow } from '../../components/common/CryptoExchangeCreateWalletRow.js'
+import { CryptoExchangeWalletListTokenRowConnected as CryptoExchangeWalletListRow } from '../../connectors/components/CryptoExchangeWalletListRowConnector.js'
+import type { GuiWallet, MostRecentWallet } from '../../types/types.js'
+import { scale } from '../../util/scaling.js'
+import { type AirshipBridge, AirshipModal } from './modalParts.js'
+
+export type StateProps = {
+  activeWalletIds: Array<string>,
+  mostRecentWallets: Array<MostRecentWallet>
+}
+
+type OwnProps = {
+  bridge: AirshipBridge<GuiWallet | Object | null>,
+  headerTitle: string,
+  wallets: Array<GuiWallet>,
+  createWalletCurrencies: Array<Object>,
+  showCreateWallet: boolean,
```

Comment:
Can we make `createWalletCurrencies` optional? Actually, can't this just be `allowedCurrencyCodes`, minus any wallet types we already have?